Add disable auto column index feature

GoldSucc opened the pull request:
<!--
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design-blazor/ant-design-blazor/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

-->

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [x] Other (about what?)

### 🔗 Related issue link

<!--
1. Describe the source of requirement, like related issue link.
-->

### 💡 Background and solution

<!--
1. Describe the problem and the scenario.
2. GIF or snapshot should be provided if includes UI/interactive modification.
3. How to fix the problem, and list final API implementation and usage sample if that is an new feature.
-->

The problem is that component initialisation order is not guaranteed so assigning column index by column context is very brittle and in case if some columns in table have nested component structure it can mess up order and incorrect indexes will be assigned which leads to broken table. With this change, there is an option to disable automatic column index assignment through context and assign indexes to columns by hand. It does not change behaviour with default parameter values so if AutoColIndex property isn't assigned to table, nothing changes. Alternative way of stabilising table is to assign columns as an array of render fragments, however it will require significant changes in API.

### 📝 Changelog

<!--
Describe changes from userside, and list all potential break changes or other risks.
--->

Risks are new exposed API which then can be difficult to rollback once it will be used in projects.

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] Changelog is provided or not needed

GoldSucc commented:
@dotnet-policy-service agree

ElderJames commented:
Thanks for contributionn @GoldSucc , there are still some improvements to be made here:

1. Please revert changes that are not relevant to this submission, such as line breaks and spaces. Because his will hinder the evaluation of the true scope of the changes.
2. Please add a attribute `[PublicApi(1.1.0)]` for the new public parameters.
3. I think it's good to add this new feature to the [generate columns(https://antblazor.com/en-US/components/table#components-table-demo-generate-columns) demo, because this would solve the issue around it, such as #4177 #3697.

GoldSucc commented:
@ElderJames I will try to make changes on weekends, having a very busy week so far

GoldSucc commented:
@ElderJames I made changes to documentation of GenerateColumns and also reverted changes to formatting, added PublicApi attribute to new public attributes

ElderJames commented on components/table/GenerateColumns.razor.cs:
```diff
@@ -56,6 +56,11 @@ protected override void BuildRenderTree(RenderTreeBuilder builder)
+                
+                if (instance != null) 
+                {
+                    instance.ColIndex = i / 3;
```
Why the ColIndex is i / 3?

GoldSucc replied:
for every column i is incremented thrice as I see in this loop, and col index should be i / 3 because of this.

ElderJames replied:
It's better to define another varibale for col index

ElderJames commented on components/table/Table.razor.cs:
```diff
@@ -69,6 +70,10 @@ public partial class Table<TItem> : AntDomComponentBase, ITable, IEqualityCompar
 
+        [Parameter]
+        [PublicApi("1.1.0")]
+        public bool AutoColIndexes { get; set; } = true;
```
Please add the comment for the parameter.

GoldSucc commented:
@ElderJames I added variable for column index as separate variable, code readability is improved now

ElderJames commented:
I think the GenerateColumns need to get the current colindex because it may be placed between other columns.

GoldSucc commented:
@ElderJames Start column index for generated columns added
